.github/workflows/codecov-external-pr.yml

```diff
@@ -14,9 +14,18 @@ jobs:
     outputs:
       coordinator: ${{ steps.filter.outputs.coordinator }}
       smart-contracts: ${{ steps.filter.outputs.smart-contracts }}
+      commit-tag: ${{ steps.get-commit-tag.outputs.COMMIT_TAG }}
     steps:
       - name: Checkout
         uses: actions/checkout@v4
+      - name: Get commit tag
+        shell: bash
+        id: get-commit-tag
+        run: |
+          echo "COMMIT_TAG=$(git rev-parse --short ${{ github.event.workflow_run.head_sha }})" >> $GITHUB_OUTPUT
```

Comment:
I assume this will return the head commit hash of the feature branch instead of the head of the temporary merge commi, and in case of "push" to main, this will return the head commit hash of `main`, right?

Reply:
Sure will do

Yes this workflow will only run on PRs made by external contributor

Previously `git rev-parse --short "$GITHUB_SHA"` would return the commit for the 'main' branch (because the only the workflow-run file in 'main' branch will run, even if you have a different version in a PR branch).

Will write comments to clarify this :)

Reply:
To clarify this is the situation for GITHUA_SHA

| Trigger Type           | New Commit on `main`           | New Commit on PR Branch                              |
|------------------------|--------------------------------|------------------------------------------------------|
| `push` / `pull_request`| Commit SHA from `main` branch  | Commit SHA from temporary merge commit (not visible in GitHub UI)    |
| `workflow_run`         | Commit SHA from `main` branch  | Commit SHA from `main` branch                        |

Reply:
added comments